practices/refactoring/readme.md

```diff
@@ -0,0 +1,93 @@
+# Refactoring
+
+Refactoring is very tightly coupled to Simple Design and TDD. TDD enables simple design, while the practice of refactoring ensures that our code bases adapt to new or changing requirements while still keeping its maintainability.
+
+## The What and The Why
+
+### The What
+
+Refactoring is a systematic process for improving code without creating new functionality.
+
+### The Why
+
+The goal of refactoring is to remediate technical debt and transform sloppy code into [clean code](/corevalues/cleancode/cleancode.md) with a simple design.
+
+#### Technical Debt and Causes
+
+Technical debt reflects the implied cost of reworking code caused by choosing a quick and easy solution instead of a better more involved approach. Technical debt is one of the main reasons you would want to refactor code, with the main culprits being:
+
+1. Business pressure
+
+2. Lack of tests
+
+3. Lack of interaction between team members
+
+4. Long-term simultaneous development in several branches
+
+5. Delayed refactoring (yes, this can lead to stacking more technical debt on top of each other)
+
+6. Lack of compliance monitoring
+
+## When to Refactor
+
+1. The rule of Three:
+    - When you’re doing something for the first time, just get it done.
+    - When you’re doing something similar for the second time, cringe at having to repeat but do the same thing anyway.
```

Comment:
this completely disregards advice given elsewhere where DRY is heavily, almost religiously advocated

also, this feels really arbitrary and there are better arbitrary times to refactor, I prefer [Red-Green-Refactor](https://youtu.be/EZ05e7EMOLM)

Reply:
refactored.